GetStream/stream-chat-android
# [4607] [V5] Populate user mentions from the backend

### 🎯 Goal

closes #4607 

### 🛠 Implementation details

Implemented fallback patch for querying members online if there are no matching mentions found in the local state user data.

### 🎨 UI Changes

No UI Changes, only behavioral changes.

### 🧪 Testing

The following tests need to be done using the following:

- The Compose app
- The XML app with `MessageComposerView`
- The XML sample app with `MessageInputView`

In the XML sample `MessageComposerView` is used by default, long press on the user icon in the channels menu to switch to using `MessageInputView`.

#### Test:

Test 1: New functionality testing

1. Open the app
2. Log in as `Jc`
3. Enter the channel `Big group`. This channel has 90 members while we only store 30, so the fact that it's bigger than the first page of channel members we fetch is important.
4. If you are testing Compose or `MessageComposerView`, filter logs for "[handleMentionSuggestions]", otherwise filter logs for "[queryMembersByUserNameContains]".
5. Type in `@` to initiate a suggestion list

Expected: You should see clear log info stating that we are querying members online for any user not found inside the first page of local state data we keep. Every user contained within the first page - which is easiest checked by typing in `@` and just checking the list -  should be fetched locally and not trigger an online member query.

Test 2: Existing functionality testing

Since we've changed the way inputs function, please validate that all previous input features work correctly as well, such as: commands, sending messages, editing messages, sending attachments, etc.

Please note that that anything leveraging `MessageComposerController` (Compose and MCV) will print to logs when it's doing local searches, while `MessageInputView` logs online searches only.

### ☑️Contributor Checklist

#### General
- [x] PR targets the ~~`develop`~~ `v5` branch

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![gif](https://media0.giphy.com/media/jgeBT8lDzAoSdwxBeV/giphy.gif?cid=ecf05e479z74lue4k4wjhlh0bsf42tea84xlumselaiuegej&rid=giphy.gif&ct=g)
